src/test/java/com/snowflake/kafka/connector/records/RecordContentTest.java

```diff
@@ -125,103 +130,98 @@ public void test() throws IOException {
+        Arguments.of(
+            "not empty value without schema",
+            new SinkRecord(topic, partition, null, null, null, content, partition)));
   }
 
-  @Test(expected = SnowflakeKafkaConnectorException.class)
-  public void testWrongValueSchema() throws IOException {
-    JsonNode data = mapper.readTree("{\"name\":123}");
-    SnowflakeRecordContent content = new SnowflakeRecordContent(data);
+  @ParameterizedTest
+  @MethodSource("invalidSchemaSource")
+  public void recordService_getProcessedRecordForSnowpipe_whenInvalidSchema_throwException(
+      String testCase, Schema schema, Object value) {
+    // given
     RecordService service = new RecordService();
+    SinkRecord record = new SinkRecord(topic, partition, null, null, schema, value, partition);
 
-    SinkRecord record =
-        new SinkRecord(
-            topic,
-            partition,
-            null,
-            null,
-            SchemaBuilder.string().name("aName").build(),
-            content,
-            partition);
-    // TODO: SNOW-215915 Fix this after stability push, if schema does not have a name
-    // There is OOM error in this test.
-    service.getProcessedRecordForSnowpipe(record);
-  }
-
-  @Test(expected = SnowflakeKafkaConnectorException.class)
-  public void testWrongValueType() {
-    RecordService service = new RecordService();
-
-    SinkRecord record =
-        new SinkRecord(
-            topic, partition, null, null, new SnowflakeJsonSchema(), "string", partition);
-    service.getProcessedRecordForSnowpipe(record);
+    // expect
+    Assertions.assertThrows(
+        SnowflakeKafkaConnectorException.class,
+        () -> service.getProcessedRecordForSnowpipe(record));
   }
 
-  @Test(expected = SnowflakeKafkaConnectorException.class)
-  public void testWrongKeySchema() throws IOException {
-    JsonNode data = mapper.readTree("{\"name\":123}");
-    SnowflakeRecordContent content = new SnowflakeRecordContent(data);
-    RecordService service = new RecordService();
-
-    SinkRecord record =
-        new SinkRecord(
-            topic,
-            partition,
+  public static Stream<Arguments> invalidSchemaSource() throws JsonProcessingException {
+    return Stream.of(
+        Arguments.of(
+            "schema not matching content",
             SchemaBuilder.string().name("aName").build(),
-            content,
-            null,
-            null,
-            partition);
-    service.putKey(record, mapper.createObjectNode());
+            new SnowflakeRecordContent(mapper.readTree("{\"name\":123}"))),
+        Arguments.of("invalid schema type", new SnowflakeJsonSchema(), "string"));
   }
 
-  @Test(expected = SnowflakeKafkaConnectorException.class)
-  public void testWrongKeyType() {
+  @ParameterizedTest
+  @MethodSource("invalidPutKeyInputSource")
+  public void recordService_putKey_whenInvalidInput_throwException(
+      String testCase, Schema schema, Object content) {
```

Comment:
Let's use `Named` arguments instead of passing a test case name. See [a stackoverflow thread](https://stackoverflow.com/questions/57892989/generating-display-names-for-parameterizedtest-in-junit-5).

Reply:
done